maven3 compatibility

When using the plugin with maven3, a NPE happens:

[ERROR] Failed to execute goal net.surguy:maven-displaylicense-plugin:0.5:displaylicense (default-cli) on project parent: Execution default-cli of goal net.surguy:maven-displaylicense-plugin:0.5:displaylicense failed. NullPointerException -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

--- a/src/main/java/net/surguy/maven/licensing/LicenseCheckerMojo.java
+++ b/src/main/java/net/surguy/maven/licensing/LicenseCheckerMojo.java
@@ -1,6 +1,7 @@
 package net.surguy.maven.licensing;
 
 import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.artifact.resolver.ArtifactCollector;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.project.MavenProject;
@@ -45,9 +46,19 @@ public class LicenseCheckerMojo extends AbstractMojo {
     @SuppressWarnings({"UnusedDeclaration"})
     private MavenMetadataSource metadataSource;
 
+    /**
+     * The Artifact Collector.
+     *
+     * @component
+     * @required
+     * @readonly
+     */
+    @SuppressWarnings({"UnusedDeclaration"})
+    private ArtifactCollector collector;
+
     public void execute() throws MojoExecutionException {
         assert project != null : "Project is null - dependency injection is not working";
-        LicenseChecker licenseChecker = new LicenseChecker(project, localRepository, metadataSource, getLog());
+        LicenseChecker licenseChecker = new LicenseChecker(project, localRepository, metadataSource, collector, getLog());
         licenseChecker.execute();
     }
 

--- a/src/main/scala/net/surguy/maven/licensing/LicenseChecker.scala
+++ b/src/main/scala/net/surguy/maven/licensing/LicenseChecker.scala
@@ -26,13 +26,12 @@ import org.apache.maven.plugin.logging.Log
  * @author Inigo Surguy
  * @created 16/01/2011 22:14
  */
-class LicenseChecker(project: MavenProject, localRepository: ArtifactRepository, metadataSource: MavenMetadataSource, log: Log) {
+class LicenseChecker(project: MavenProject, localRepository: ArtifactRepository, metadataSource: MavenMetadataSource, collector: ArtifactCollector, log: Log) {
   val licenseExtractor = new LicenseExtractor()
 
   def execute() {
     val artifacts:Set[Artifact] = project.getDependencies.map(d => toArtifact(d.asInstanceOf[Dependency])).collect{case a: Artifact => a}.toSet
 
-    val collector = new DefaultArtifactCollector()
     val listener = new DependencyTreeResolutionListener(new ConsoleLogger(Logger.LEVEL_WARN, "Resolution"))
     collector.collect(artifacts, project.getArtifact(), project.getManagedVersionMap(),
       localRepository, project.getRemoteArtifactRepositories(), metadataSource, new ScopeArtifactFilter("compile"),

